Bottom: 22e3b80e1c1bc2fd08f8b277615eb4de224b84af
Top:    0b025914f7b10fb4e46fd87593f2d395c83ef82d
Author: Alexandre Oliva <aoliva@redhat.com>
Date:   2016-03-02 01:14:51 -0300

introduce callback to get type char

---

diff --git a/include/gcc-cp-fe.def b/include/gcc-cp-fe.def
index 4a6fb7f..3571834 100644
--- a/include/gcc-cp-fe.def
+++ b/include/gcc-cp-fe.def
@@ -327,6 +327,11 @@ GCC_METHOD2 (gcc_type, int_type,
 	     int /* bool */,		   /* Argument IS_UNSIGNED.  */
 	     unsigned long)                /* Argument SIZE_IN_BYTES.  */
 
+/* Return the 'char' type, distinct from both 'signed char' and
+   'unsigned char' returned by the above.  */
+
+GCC_METHOD0 (gcc_type, char_type)
+
 /* Return a floating point type with the given properties.  */
 
 GCC_METHOD1 (gcc_type, float_type,
diff --git a/libcc1/libcp1plugin.cc b/libcc1/libcp1plugin.cc
index 7032ac0..abe42a0 100644
--- a/libcc1/libcp1plugin.cc
+++ b/libcc1/libcp1plugin.cc
@@ -1263,6 +1263,12 @@ plugin_int_type (cc1_plugin::connection *self,
 }
 
 gcc_type
+plugin_char_type (cc1_plugin::connection *self)
+{
+  return convert_out (char_type_node);
+}
+
+gcc_type
 plugin_float_type (cc1_plugin::connection *,
 		   unsigned long size_in_bytes)
 {
